src/mon/PGMap.cc

```diff
@@ -1754,24 +1754,17 @@ void PGMap::generate_test_instances(list<PGMap*>& o)
   }
 }
 
-void PGMap::get_filtered_pg_stats(const string& state, int64_t poolid, int64_t osdid,
+void PGMap::get_filtered_pg_stats(__u32 state, int64_t poolid, int64_t osdid,
```

Comment:
`__u32` is a linux specific type, maybe we can try to use `uint32_t` instead? i see the underlying type of `pg_state_t::state` is `__u32`. but we can be more portable by using these ANSI types.